readthedocs/organizations/views/base.py

```diff
@@ -20,27 +18,14 @@
     Team,
     TeamMember,
 )
-
-
-class CheckOrganizationsEnabled:
```

Comment:
I'm confused why this was reimplemented as a decorator instead of keeping it at a mixin?

Reply:
To me, it doesn't really seem like a decorator change here is necessary at all. The view that we wanted to implement was just the redirect view. With that implemented, the magic `-` URLs all seem redundant. I'd stick to using the redirect view too, it's nice an explicit where a magic URL is not.

Reply:
It's another way to implement the same thing. But I wanted to have real URL paths instead of query string arguments prone to typos.

With this pattern, you simply swap the slug for a `-` whenever you want to make it generic/sharable. With the other pattern, you have more manual work and a new margin for error.

Reply:
We're not talking about using these URLs in many places, so I don't think that needs to a concern in our implementation. The use case for this so we can add a link to some of the pages from documentation and support response templates.

Reply:
I agree with @agjohnson that this PR adds a lot of complexity to save a small amount of time when generating one of these URLs? It seems like a poor trade off. I'd much prefer an explicit redirect view that is scalable and easy to understand, than a quite complex decorator implementation that introduces ambiguity to our URL patterns that aren't consistent across various types of URLs (eg. the magic `-` slug wouldn't work for many places where we use slugs).